docs/metrics.md

```diff
@@ -0,0 +1,422 @@
+# Pulse metrics
+
+A summary of the metrics the Pulse add-on will record.
+
+
+## Definitions
+
+* **Report** - _TBD_
+* **Page** - _TBD_
+* **Session** - _TBD_
+* **Engagement** - _TBD_
+
+Also _TBD_: ensuring that these terms are correctly used in the document below.
+
+
+## Data analysis
+
+_TBD_
+
+
+## Data collection
+
+A Pulse report is initiated when the user chooses to begin a survey, either on their own initiative or by prompting. Once the survey is completed, that data is passed on to the add-on, which augments it with information about the browser, the page from which the survey was begun, and the session to which that page belongs.
+
+This payload is then sent to the Test Pilot add-on, which forwards it to [Unified Telemetry](https://wiki.mozilla.org/Unified_Telemetry) and [Ping Centre](https://github.com/mozilla/ping-centre) for analysis.
+
+
+### Properties
+
+Details about each property in the payload.
+
+
+#### Survey data
+
+Data gathered from the user in the short survey at the beginning of the report lifecycle.
+
+
+##### `type`
+
+The reason for the submission. String, possible values:
+
+* `random` - if the user was randomly prompted.
+* `user` - if the user initiated the report by the WebExtension page action.
+
+
+##### `sentiment`
+
+The user's feeling about the current page, scored on a five-point Likert scale. Number, between `1` and `5`, inclusive.
+
+
+##### `reason`
+
+The reason the user feels the way they indicated in the `sentiment`. String, possible value _TBD_.
+
+
+##### `details`
+
+Additional details about why the user feels the way they indicated in `sentiment` and `reason`, entered into an open textarea. String or `null`.
+
+  "details": null,
+
+
+#### Browser data
+
+Characteristics of the browser, unrelated to the page or session.
+
+
+##### `telemetryId`
+
+The user's telemetry ID, retrieved from `toolkit.telemetry.cachedClientID`. String, UUID.
+
+
+##### `platform`
+
+The name of the operating system. String, one of the [`OS_TARGET`](https://developer.mozilla.org/docs/Mozilla/Developer_guide/Build_Instructions/OS_TARGET) values.
+
+
+##### `language`
+
+The user's preferred language. String, gotten from [`navigator.language`](https://developer.mozilla.org/docs/Web/API/NavigatorLanguage/language).
+
+
+##### `version`
+
+The version of Firefox, gotten from [`system.version`](https://developer.mozilla.org/Add-ons/SDK/High-Level_APIs/system#version).
+
+
+##### `channel`
+
+The Firefox release channel. String, one of `nightly`, `developer`, `beta`, or `release`.
+
+
+##### `addons`
+
+A list of IDs of installed add-ons. Array.
+
+
+##### `adBlocker`
+
+Does the user have an add-on installed that is known to block ads? Boolean.
+
+
+##### `openTabs`
+
+The count of open tabs across all windows. Number.
+
+
+##### `openWindows`
+
+The number of open windows across the browser. Number.
+
+
+##### `hangs`
```

Comment:
I recommend splitting hangs for child and main process. The metric/filter for this in the past was "hangs over 100ms per second". This could also be moved to session, as it can then be normalized by the session length